internal/database/codeowners.go

```diff
@@ -0,0 +1,83 @@
+package database
+
+import (
+	"context"
+	"database/sql"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/sourcegraph/sourcegraph/internal/api"
+	"github.com/sourcegraph/sourcegraph/internal/database/basestore"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+
+	codeownerspb "github.com/sourcegraph/sourcegraph/internal/own/codeowners/proto"
+)
+
+// CodeownersStore currently allows to store only the most recent version of CODEOWNERS
+// file for main branch of given repository.
+type CodeownersStore interface {
+	basestore.ShareableStore
+	// Put updates the newest CODEOWNERS file for given repository's HEAD.
+	PutHead(context.Context, api.RepoName, *codeownerspb.File) error
+	// GetHead returns the CODEOWNERS file associated with
+	// or nil, if there is none.
+	GetHead(ctx context.Context, repoName api.RepoName) (*codeownerspb.File, error)
+}
+
+var _ CodeownersStore = (*codeownersStore)(nil)
+
+type codeownersStore struct {
+	*basestore.Store
+}
+
+func (s *codeownersStore) PutHead(ctx context.Context, repoName api.RepoName, f *codeownerspb.File) error {
+	b, err := proto.Marshal(f)
```

Comment:
I would name it a bit less abstract, e.g.:

```suggestion
	codeownersData, err := proto.Marshal(f)
```